### PR TITLE
feat(ui): add workflow usage

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

### CODE REVIEW

File: sdk/workflow.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ type Workflow struct {
 	Joins        []WorkflowNodeJoin `json:"joins,omitempty" db:"-" cli:"-"`
 	Groups       []GroupPermission  `json:"groups,omitempty" db:"-" cli:"-"`
 	Permission   int                `json:"permission,omitempty" db:"-" cli:"-"`
+	Usage        Usage              `json:"usage" db:"-"`
```
Comment (yesnault) [inline sdk/workflow.go]: cli:"-"

File: ui/src/assets/i18n/en.json [hunk 2]
```diff
@@ -393,8 +393,8 @@
   "project_ssh_error" : "Project ssh key must have a name",
   "project_update_name_error" : "Project name is mandatory and must respect the alphanumeric pattern ([a-zA-Z0-9]*)",
   "project_variable_list_title" : "List of project variables: ",
-  "project_variable_form_title" : "Add a new variable: ",
-  "project_workflows_list" : "List of workflows in the project: ",
+  "project_variable_form_title" : "Add a new variable : ",
```
Comment (yesnault) [inline ui/src/assets/i18n/en.json]: no space before in en.

File: ui/src/assets/i18n/en.json [hunk 3]
```diff
@@ -393,8 +393,8 @@
...
+  "project_workflows_list" : "List of workflows in the project : ",
```
Comment (yesnault) [inline ui/src/assets/i18n/en.json]: no space before in en.

File: ui/src/assets/i18n/fr.json [hunk 4]
```diff
@@ -395,6 +395,7 @@
   "project_variable_list_title" : "Liste des variables du projet : ",
   "project_variable_form_title" : "Ajouter une nouvelle variable : ",
   "project_pipelines_list" : "Liste des workflows du projet : ",
```
Comment (yesnault) [inline ui/src/assets/i18n/fr.json]: liste des pipelines

File: engine/api/workflow.go [hunk 5]
```diff
@@ -36,17 +40,60 @@ func (api *API) getWorkflowHandler() Handler {
...
+
+		if errTxc := tx.Commit(); errTxc != nil {
+			return sdk.WrapError(errTxc, "getWorkflowHandler> Cannot commit transaction for workflow %s", name)
+		}
+
 		return WriteJSON(w, r, w1, http.StatusOK)
 	}
 }
 
+func loadWorfklowUsage(db gorp.SqlExecutor, wf sdk.Workflow) (sdk.Usage, error) {
```
Comment (sguiheux) [inline engine/api/workflow.go]: in dao file?

File: engine/api/application/application_pipeline.go [hunk 6]
```diff
@@ -118,6 +118,25 @@ func GetAllPipelinesByID(db gorp.SqlExecutor, applicationID int64) ([]sdk.Applic
 	return appPipelines, nil
 }
 
+// LoadByWorkflow loads applications from database
+func LoadByWorkflow(db gorp.SqlExecutor, wf sdk.Workflow) ([]sdk.Application, error) {
```
Comment (fsamin) [inline engine/api/application/application_pipeline.go]: move this in the `dao.go` file

Comment (fsamin) [inline engine/api/application/application_pipeline.go]: only workflow_id is usefull as argument

Comment (fsamin) [inline engine/api/application/application_pipeline.go]: please write a unit test for this dao function

File: engine/api/environment/environment.go [hunk 7]
```diff
@@ -144,6 +144,25 @@ func LoadByPipelineName(db gorp.SqlExecutor, projectKey, pipName string) ([]sdk.
 	return envs, nil
 }
 
+// LoadByWorkflow loads environements from database
+func LoadByWorkflow(db gorp.SqlExecutor, wf sdk.Workflow) ([]sdk.Environment, error) {
```
Comment (fsamin) [inline engine/api/environment/environment.go]: id is enough

File: engine/api/pipeline/pipeline.go [hunk 8]
```diff
@@ -106,6 +106,24 @@ func LoadPipelineByID(db gorp.SqlExecutor, pipelineID int64, deep bool) (*sdk.Pi
 	return &p, nil
 }
 
+// LoadByWorkflow loads pipelines from database
+func LoadByWorkflow(db gorp.SqlExecutor, wf sdk.Workflow) ([]sdk.Pipeline, error) {
```
Comment (fsamin) [inline engine/api/pipeline/pipeline.go]: id is enough

File: engine/api/workflow.go (hunk 5 above)
Comment (fsamin) [inline engine/api/workflow.go]: id is enough; 
i wonder if this should be in workflow package...

File: sdk/workflow.go [hunk 9]
```diff
@@ -19,6 +19,7 @@ type Workflow struct {
 	Joins        []WorkflowNodeJoin `json:"joins,omitempty" db:"-" cli:"-"`
 	Groups       []GroupPermission  `json:"groups,omitempty" db:"-" cli:"-"`
 	Permission   int                `json:"permission,omitempty" db:"-" cli:"-"`
+	Usage        Usage              `json:"usage" db:"-" cli:"-"`
```
Comment (fsamin) [inline sdk/workflow.go]: a pointer to Usage and a omitempty tag would be nice

File: engine/api/workflow.go [hunk 10]
```diff
@@ -36,17 +40,60 @@ func (api *API) getWorkflowHandler() Handler {
...
+			return sdk.WrapError(errTx, "getWorkflowHandler> Cannot start transaction for workflow %s", name)
+		}
+		defer tx.Rollback()
+
+		w1, err := workflow.Load(tx, api.Cache, key, name, getUser(ctx))
 		if err != nil {
 			return err
 		}
+
+		if withUsage {
```
Comment (fsamin) [inline engine/api/workflow.go]: please write a handler test for this

File: engine/api/workflow.go [hunk 11]
```diff
@@ -36,17 +40,60 @@ func (api *API) getWorkflowHandler() Handler {
...
-		w1, err := workflow.Load(api.mustDB(), api.Cache, key, name, getUser(ctx))
+		tx, errTx := api.mustDB().Begin()
+		if errTx != nil {
+			return sdk.WrapError(errTx, "getWorkflowHandler> Cannot start transaction for workflow %s", name)
+		}
+		defer tx.Rollback()
+
+		w1, err := workflow.Load(tx, api.Cache, key, name, getUser(ctx))
 		if err != nil {
 			return err
```
Comment (fsamin) [inline engine/api/workflow.go]: wrap this error

**Pull request merged**